### ISSUE TITLE
Interacting with WebView on Android is flaky

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#244]
Flutter's semantics is not replicated by the Android accessibility framework during `flutter drive`.

In practice, the biggest pain that this problem creates is the inability to interact with WebViews on Android.

### flutter run

When the app is run normally (i.e through `main.dart`) with `flutter run`, everything works fine – Android's native accessibility tree matches Flutter's semantics tree.

For example, when the currently displayed screen is the main screen: 
<img width="468" alt="Screenshot 2022-09-06 at 1 26 59 AM" src="https://user-images.githubusercontent.com/40357511/188520320-931f2ac1-00ef-457a-9f82-3aa42448f020.png">

and I use [http](https://httpie.io) to get the native Android views, I get the following output:

<details>
  <summary>http output</summary>

    $ http POST localhost:8081/getNativeWidgets className=android.widget.Button
    HTTP/1.1 200 OK
    connection: keep-alive
    content-length: 2593
    content-type: text/plain; charset=utf-8
    [
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": null,
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": null,
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": null,
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": null,
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open loading screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open location screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open notifications screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open overlay screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open permissions screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open scrolling screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": "Open webview screen",
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        },
        {
            "applicationPackage": "pl.leancode.maestro.example",
            "childCount": 0,
            "children": [],
            "className": "android.widget.Button",
            "contentDescription": null,
            "enabled": true,
            "focused": false,
            "resourceName": "",
            "text": null
        }
    ]

</details>

Also, `evilmaestro hierarchy` prints the same.

### flutter drive

When the app is run in an integration test with `flutter drive`, it looks like as if Flutter's semantics tree is not sent to the Android side.

To verify this behavior, I do `maestro drive -t integration_test/no_op_test.dart` in `maestro_test`'s example app directory and then I use the [http](https://httpie.io/) tool to see how Flutter's semantics tree got translated into native Android view hierarchy.

```
$ http POST localhost:8081/getNativeWidgets className=android.widget.Button
HTTP/1.1 200 OK
connection: keep-alive
content-length: 2
content-type: text/plain; charset=utf-8

[]
```

I get nothing in response.

`evilmaestro hierarchy` also fails in this scenario (because it just walks the accessibility tree, which apparently is non-existent. 

**What's interesting** is that if TalkBack is enabled, then the semantics tree is sent from Flutter to Android even during `flutter drive`. Demo below. I run the test with `maestro drive -t integration_test/webview_test.dart`.

https://user-images.githubusercontent.com/40357511/188608198-eb316143-c797-45ae-b855-adc638ab1c3b.mp4

### More context

- The first investigation that was conducted in #218
- #235
- #239

**Comment by bartekpacia** [comment]
I can't believe my eyes, but a WebView test just passed on Android.

https://user-images.githubusercontent.com/40357511/200897638-c04d1a03-3a3f-49cd-8466-c4b47fccd3b2.mov

**Comment by jagadeeshpurplle** [comment]
Hi @bartekpacia, i just want to ask that you have implemented the flutter webview using package https://pub.dev/packages/webview_flutter right?

**Comment by bartekpacia** [comment]
@jagadeeshpurplle, yes we're using that package. You can see it [here](https://github.com/leancodepl/patrol/blob/master/packages/patrol/example/lib/webview_screen.dart).

**Comment by akagupta9** [comment]
any updates on it?

**Comment by azeunkn0wn** [comment]
Still no progress for this issue? :(

Is there an alternative? like can you make the app blindly tap on a specific (X, Y) location of the screen and type text on it?
I'm new to Integration testing, and I'm not sure if functions like that exist.

**Comment by bartekpacia** [comment]
Possible _temporary_ workarounds:
- subscribe to "WebView finished loading" callback and then progress with the test
- add a dummy 1s or 2s wait (with pumping) after webview is loaded

**Comment by bartekpacia** [comment]
This was fixed by #1398 on newer Android versions. It works _quite reliably_ since Android 10 (API 29).
Unfortunately, on older versions it still fails 100% of the time. This might be related to how Flutter integrates with platform views on Android. 
[Texture Layer Hybrid Composition](https://github.com/flutter/flutter/wiki/Texture-Layer-Hybrid-Composition) has much better performance since Android 10.
Under certain circumstances, fallbacks to [Hybrid Composition](https://github.com/flutter/flutter/wiki/Hybrid-Composition) or [Virtual Display](https://github.com/flutter/flutter/wiki/Virtual-Display) might occur.
This requires further, deep investigation.

**Comment by bartekpacia** [comment]
Renamed to reflect current situation.

**Comment by bartekpacia** [comment]
Reported the bug to Flutter - see https://github.com/flutter/flutter/issues/130872.
